build_tools/scripts/update_submodules.py

```diff
@@ -0,0 +1,224 @@
+    if args.update_build_files is None:
+      print("Will update build files from TensorFlow",
+            "because --update_build_files not specified")
+      args.update_build_files = True
+  if args.llvm_commit == "KEEP":
+    print("Not updating LLVM (--llvm_commit == 'KEEP')")
+  else:
+    print("\n*** Updating LLVM to", args.llvm_commit, "***")
+    update_submodule(args.llvm, args.llvm_commit)
+
+  # Update build files.
+  if not args.update_build_files:
+    print("Not updating build files (--update_build_files not specified)")
+  else:
+    print("\n*** Updating BUILD.bazel files ***")
+    update_build_files_from_tensorflow(args.repo, args.tensorflow)
+
+
+def execute(args, cwd, capture_output=False, silent=False, **kwargs):
+  if not silent:
+    print("+", " ".join(args), "  [from %s]" % cwd)
+  if capture_output:
+    return subprocess.check_output(args, cwd=cwd, **kwargs)
+  else:
+    return subprocess.check_call(args, cwd=cwd, **kwargs)
+
+
+def get_commit(path, rev="HEAD"):
+  return execute(["git", "rev-parse", rev],
+                 cwd=path,
+                 silent=True,
+                 capture_output=True).decode("ISO-8859-1").strip()
+
+
+def update_submodule(path, commit, tracking="origin/master"):
+  # Fetch.
+  execute(["git", "fetch"], cwd=path)
+  # Determine commit.
+  if commit == "REMOTE":
+    commit = get_commit(path, rev=tracking)
+    print("Resolved remote commit:", commit)
+
+  # Rebase to commit (will fail if not fast-forward).
+  execute(["git", "rebase", commit], cwd=path)
+
+
+def find_tensorflow_llvm_commit(tensorflow_path):
+  # TensorFlow keeps its commit in workspace.bzl on a line like:
+  # LLVM_COMMIT = "..."
+  # Yeah. This is how we do it.
+  workspace_path = os.path.join(tensorflow_path, "tensorflow", "workspace.bzl")
+  pattern_text = r"""\s*LLVM_COMMIT\s*=\s*"(.+)"\s*"""
+  pattern = re.compile(pattern_text, flags=re.MULTILINE)
+  for line in open(workspace_path, "r", encoding="UTF-8"):
+    m = re.match(pattern, line)
+    if m:
+      return m.group(1)
+
+  print("ERROR: Could not find LLVM commit in %s." % workspace_path)
+  print("Request an explicit commit via --llvm_commit (and file a bug)")
+  print("Expected pattern match for:", pattern_text)
+  sys.exit(1)
+
+
+def update_build_files_from_tensorflow(repo_path, tensorflow_path):
+  src_llvm_build = os.path.join(tensorflow_path, "third_party", "llvm",
+                                "llvm.autogenerated.BUILD")
+  # NOTE(laurenzo): These will probably move upstream.
+  src_mlir_build = os.path.join(tensorflow_path, "third_party", "mlir", "BUILD")
+  src_mlir_test_build = os.path.join(tensorflow_path, "third_party", "mlir",
+                                     "test.BUILD")
+  overlay_path = os.path.join(repo_path, "build_tools", "bazel",
+                              "third_party_import", "llvm-project", "overlay")
+  # Both TensorFlow and IREE are copyright Google with the same license,
```

Comment:
We could also exclude this directory from the license check if that seems better than adding the license.

Reply:
Yes, let's do that. How?

Reply:
I'll send you a CL